frontend/simple/js/backend/hapi.js

```diff
@@ -54,11 +69,28 @@ const primus = pubsub({
 export class HapiBackend extends Backend {
   constructor () {
     super()
+    this.subscriptions = []
     primus.on('reconnected', () => {
-      var subs = this.subscriptions()
+      var subs = this.subscriptions
       console.log('websocket connection re-established. re-joining:', subs)
       subs.forEach(contractId => primus.sub(contractId))
     })
+    var updateSubs = async ({data}) => {
```

Comment:
A comment here about intent of this function would be good as well as what is expected in terms of properties on this data object.